web/ui/src/common/miscUtils.js

```diff
@@ -183,6 +192,37 @@
                 }, 0);
 
                 return childCount;
+            },
+
+            validateRAMLimit(limitStr, max=Infinity){
+
+                var isPercent = (limitStr.endsWith("%"));
+
+                // if this is a percent, ensure its between 1 and 100
+                if(isPercent){
+                    let val = +limitStr.slice(0, -1);
+                    if(val > 100){
+                        return "RAM Limit cannot exceed 100%";
```

Comment:
Assuming these are error messages to be displayed to the user, do they need to be localized?